numpy/core/tests/test_scalarmath.py

```diff
@@ -95,6 +85,11 @@ def test_array_scalar_ufunc_equivalence(op, arr1, arr2):
         comp_ops = {operator.ge, operator.gt, operator.le, operator.lt}
         if op in comp_ops and (np.isnan(scalar1) or np.isnan(scalar2)):
             pytest.xfail("complex comp ufuncs use sort-order, scalars do not.")
+    if op == operator.pow and arr2 == 2:
+        # array**2 special case can have different result dtype
+        # (Other powers may have issues also, but are not hit here.)
+        # TODO: It would be nice to resolve this issue.
+        pytest.skip("array**2 can have incorrect/weird result dtype")
```

Comment:
Seems to complicated to figure out... But I think all weird cases are `arr**2`.  It looks like there should be more bad places when:
* The operation is in-place (but in that case it is maybe OK, since it gets cast back anyway)
* Temporary elision kicks in.  In that case it seems like promotion may differ for mainly for 0-D arrays (this is rather special).  (Which also may be inherited by scalars)

Neither are new issues.